phoneNumber.js

```diff
@@ -30,6 +36,17 @@ console.log(testPhoneNumber('(206) 33-4444')); // should return false, missing a
 // the phone number.
 // Returns an object in the format {areaCode, phoneNumber}
 
+function parsePhoneNumber(phoneString){
+    const result = globalRegex.exec(phoneString);
```

Comment:
Okay this is really tricky I'll admit, because the `exec()` method is 'iterable' on the Regex object you created for the globalRegex. That means that when you call 2 phone number checks, it is hitting the same global regex pattern twice, but the second time around it basically thinks it's already iterated everything there.

SO one way around this is to use the global value you set, but make a brand new Regex object inside the scope of the function so that it does not persist, and ensure that the data you parse is only relevant to the latest call's parameters . . . something like this:
```js
function parsePhoneNumber(phoneString){
    const newRegex = new RegExp(globalRegex)
    const result = newRegex.exec(phoneString);
    const areaCode = result[1];
    const phoneNumber = `${result[2]}${result[3]}`;
    return {areaCode:`${areaCode}`, phoneNumber:`${phoneNumber}`};
};
```

Reply:
and since I just want to return the object itself, may as well skip setting it to a variable, it's kind of just an extra step.